src/Plugin/Field/FieldWidget/Select2Widget.php

```diff
@@ -42,7 +42,7 @@ public function settingsForm(array $form, FormStateInterface $form_state) {
       '#description' => $this->t("Define a width for the select2 field. It can be either 'element', 'style', 'resolve' or any possible CSS value. E.g. 500px, 50%, 200em. See the <a href='https://select2.org/appearance#container-width'>select2 documentation</a> for further explanations."),
       '#required' => TRUE,
       '#size' => '10',
-      '#pattern' => "^(\d+(cm|mm|in|px|pt|pc|em|ex|ch|rem|vm|vh|vmin|vmax|\%)|element|style|resolve|auto|initial|inherit)$",
+      '#pattern' => "\d+(cm|mm|in|px|pt|pc|em|ex|ch|rem|vm|vh|vmin|vmax|%)|element|style|resolve|auto|initial|inherit",
```

Comment:
`\d+` - does not accept decimal separator (`.`)

```suggestion
      '#pattern' => "([0-9]*\.[0-9]+|[0-9]+)(cm|mm|in|px|pt|pc|em|ex|ch|rem|vm|vh|vmin|vmax|%)|element|style|resolve|auto|initial|inherit",
```